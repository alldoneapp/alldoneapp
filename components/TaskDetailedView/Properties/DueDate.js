import React from 'react'
import { StyleSheet, Text, View } from 'react-native'
import Icon from '../../Icon'
import styles, { colors } from '../../styles/global'
import DueDateButton from '../../UIControls/DueDateButton'
import { translate } from '../../../i18n/TranslationService'
import { setTaskDueDate, setTaskToBacklog } from '../../../utils/backends/Tasks/tasksFirestore'

export default function DueDate({ projectId, task, disabled }) {
    const saveTaskDueDate = (taskToUpdate, dateTimestamp, isObservedTabActive) => {
        // In this context, isObservedTabActive is always false as we are not in the "Observed" tab.
        setTaskDueDate(projectId, taskToUpdate.id, dateTimestamp, taskToUpdate, false, null)
    }

    const setTaskToBacklogInDetailedView = (taskToUpdate, isObservedTabActive) => {
        // In this context, isObservedTabActive is always false.
        setTaskToBacklog(projectId, taskToUpdate.id, taskToUpdate, false, null)
    }

    return (
        <View style={localStyles.container}>
            <View style={{ marginRight: 8 }}>
                <Icon name="calendar" size={24} color={colors.Text03} />
            </View>
            <Text style={[styles.subtitle2, { color: colors.Text03 }]}>{translate('Reminder')}</Text>
            <View style={{ marginLeft: 'auto' }}>
                <DueDateButton
                    projectId={projectId}
                    task={task}
                    disabled={disabled}
                    saveDueDateBeforeSaveTask={saveTaskDueDate}
                    setToBacklogBeforeSaveTask={setTaskToBacklogInDetailedView}
                />
            </View>
        </View>
    )
}

const localStyles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'row',
        maxHeight: 56,
        minHeight: 56,
        height: 56,
        paddingLeft: 8,
        paddingVertical: 8,
        alignItems: 'center',
    },
})
