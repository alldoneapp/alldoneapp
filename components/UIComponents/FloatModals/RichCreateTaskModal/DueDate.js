import React from 'react'
import { StyleSheet } from 'react-native'
import Hotkeys from 'react-hot-keys'

import Button from '../../../UIControls/Button'
import { colors } from '../../../styles/global'
import { execShortcutFn } from '../../../../utils/HelperFunctions'

export default function DueDate({ showDueDate, disabled }) {
    return (
        <Hotkeys
            keyName={'alt+r'}
            onKeyDown={(sht, event) => {
                execShortcutFn(this.dueDateRef, showDueDate, event)
            }}
            filter={e => true}
            disabled={disabled}
        >
            <Button
                ref={ref => (this.dueDateRef = ref)}
                icon={'calendar'}
                iconColor={colors.Text04}
                buttonStyle={localStyles.buttonsStyle}
                onPress={showDueDate}
                disabled={disabled}
                shortcutText={'R'}
                forceShowShortcut={true}
            />
        </Hotkeys>
    )
}

const localStyles = StyleSheet.create({
    buttonsStyle: {
        backgroundColor: colors.Secondary200,
        marginRight: 4,
    },
})
