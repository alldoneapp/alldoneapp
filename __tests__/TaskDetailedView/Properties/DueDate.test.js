/**
 * @jest-environment jsdom
 */

import React from 'react'
import DueDate from '../../../components/TaskDetailedView/Properties/DueDate'

import renderer from 'react-test-renderer'

describe('DueDate component', () => {
    describe('DueDate snapshot test', () => {
        it('should render correctly', () => {
            const tree = renderer
                .create(<DueDate projectId={'-Asd'} task={{ id: '-Sda', name: 'My task', dueDate: 1556028000 }} />)
                .toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})
