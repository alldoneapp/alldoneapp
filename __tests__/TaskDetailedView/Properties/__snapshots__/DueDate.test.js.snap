// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DueDate component DueDate snapshot test should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "maxHeight": 56,
      "paddingLeft": 11,
      "paddingVertical": 8,
    }
  }
>
  <View
    style={
      Object {
        "marginRight": 11,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#8A94A6",
            "fontFamily": "alldone",
            "fontSize": 22,
            "opacity": 1,
          }
        }
      >
        
      </Text>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#04142F",
          "fontFamily": "Roboto-Medium",
          "fontSize": 14,
          "letterSpacing": 0.16,
          "lineHeight": 22,
        },
        Object {
          "color": "#8A94A6",
        },
      ]
    }
  >
    Reminder
  </Text>
  <View
    style={
      Object {
        "marginLeft": "auto",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "transparent",
          "borderColor": "#C6CDD2",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 1,
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "height": 40,
          "justifyContent": "center",
          "maxHeight": 40,
          "minHeight": 40,
          "opacity": 1,
          "paddingLeft": 12,
          "paddingRight": 16,
          "paddingVertical": 8,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "#8A94A6",
              "fontFamily": "alldone",
              "fontSize": 24,
              "opacity": 1,
            }
          }
        >
          
        </Text>
      </View>
      <Text
        numberOfLines={1}
        style={
          Array [
            Array [
              Object {
                "alignSelf": "center",
                "color": "#FFFFFF",
                "flexWrap": "nowrap",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "letterSpacing": 0.8,
                "lineHeight": 16,
                "margin": 0,
                "paddingHorizontal": 0,
                "paddingVertical": 0,
              },
              Object {
                "color": "#8A94A6",
              },
              Object {},
            ],
            Object {
              "marginLeft": 12,
            },
          ]
        }
      >
        18.01.70
      </Text>
    </View>
  </View>
</View>
`;
